RecoEgamma/EgammaElectronAlgos/src/PixelHitMatcher.cc

```diff
@@ -8,13 +8,83 @@
 #include "DataFormats/DetId/interface/DetId.h"
 #include "DataFormats/GeometryCommonDetAlgo/interface/PerpendicularBoundPlaneBuilder.h"
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "RecoEgamma/EgammaElectronAlgos/interface/ElectronUtilities.h"
+#include "TrackingTools/TrajectoryParametrization/interface/GlobalTrajectoryParameters.h"
+#include "TrackingTools/DetLayers/interface/rangesIntersect.h"
+#include "DataFormats/GeometryVector/interface/VectorUtil.h"
+#include "DataFormats/Math/interface/normalizedPhi.h"
+#include "RecoEgamma/EgammaElectronAlgos/interface/ElectronUtilities.h"
+#include "TrackingTools/TrajectoryParametrization/interface/GlobalTrajectoryParameters.h"
+#include "TrackingTools/DetLayers/interface/rangesIntersect.h"
+#include "DataFormats/GeometryVector/interface/VectorUtil.h"
+#include "TrackingTools/TransientTrackingRecHit/interface/TransientTrackingRecHit.h"
+#include "DataFormats/Math/interface/normalizedPhi.h"
 
 #include <typeinfo>
 #include <bitset>
 
+namespace std {
+  template <>
+  struct hash<std::pair<const GeomDet *, GlobalPoint> > {
```

Comment:
this is now too inconspicuous.
Would it work in this context to move this hash to the anonymous namespace and add that as a 3rd template parameter in the PixelHitMatcher::operator unordered_map?